src/instance.rs

```diff
@@ -887,10 +876,7 @@ where
     /// # }
     /// ```
     pub fn new(py: Python<'_>, value: impl Into<PyClassInitializer<T>>) -> PyResult<Py<T>> {
-        let initializer = value.into();
-        let obj = initializer.create_cell(py)?;
-        let ob = unsafe { Py::from_owned_ptr(py, obj as _) };
-        Ok(ob)
+        Bound::new(py, value).map(Bound::unbind)
```

Comment:
Nice! Removing unsafe by reusing existing code always make me happy

Reply:
Agreed, a lot of old PyO3 code didn't do enough of that kind of reuse (I was guilty of this too).